SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialGDKFunctionalTestsModule.cpp

```diff
@@ -12,8 +17,50 @@ DEFINE_LOG_CATEGORY(LogSpatialFunctionalTest);
 
 IMPLEMENT_MODULE(FSpatialGDKFunctionalTestsModule, SpatialGDKFunctionalTests);
 
-void FSpatialGDKFunctionalTestsModule::StartupModule() {}
+void FSpatialGDKFunctionalTestsModule::StartupModule()
+{
+	// Allow Spatial Plugin to stop PIE after Automation Manager completes the tests
+	IAutomationControllerModule& AutomationControllerModule =
+		FModuleManager::LoadModuleChecked<IAutomationControllerModule>(TEXT("AutomationController"));
+	IAutomationControllerManagerPtr AutomationController = AutomationControllerModule.GetAutomationController();
+	AutomationController->OnTestsComplete().AddLambda([]() {
+		// Make sure to clear the snapshot in case something happened with Tests (or they weren't ran properly).
+		ASpatialFunctionalTest::ClearAllTakenSnapshots();
+
+#if ENGINE_MINOR_VERSION < 25
+		if (GetDefault<USpatialGDKEditorSettings>()->bStopPIEOnTestingCompleted && GEditor->EditorWorld != nullptr)
+#else
+		if (GetDefault<USpatialGDKEditorSettings>()->bStopPIEOnTestingCompleted && GEditor->IsPlayingSessionInEditor())
+#endif
+		{
+			GEditor->EndPlayMap();
+		}
+	});
+
+	FSpatialGDKEditorModule& GDKEditorModule = FModuleManager::LoadModuleChecked<FSpatialGDKEditorModule>(TEXT("SpatialGDKEditor"));
+	GDKEditorModule.OverrideSettingsForTestingDelegate.AddLambda([](UWorld* World, const FString& MapName) {
+		FSpatialGDKFunctionalTestsModule::OverrideSettingsForTesting(World, MapName);
+	});
+}
 
 void FSpatialGDKFunctionalTestsModule::ShutdownModule() {}
 
+void FSpatialGDKFunctionalTestsModule::OverrideSettingsForTesting(UWorld* World, const FString& MapName)
```

Comment:
Looks like this only manages loading from snapshot, should probably update the name?

Reply:
Unless I misunderstand how this relates to settings